TESTS/netsocket/dns/main.cpp

```diff
@@ -160,7 +160,7 @@ static void net_bringup()
     net = NetworkInterface::get_default_instance();
     nsapi_error_t err = net->connect();
     TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, err);
-    printf("MBED: IP address is '%s'\n", net->get_ip_address());
+    printf("MBED: IP address is '%s'\n", net->get_ip_address() ? net->get_ip_address() : "null");
```

Comment:
There should be an IP address after successful connect. Could you check why it is not returned?

Reply:
After connect there might not be an ip address available from the modem. I Added retry logic to get_ip_address() in second commit. Now it works. Good catch!